pkg/storage/mysql/mysqlstore.go

```diff
@@ -101,26 +98,21 @@ func (p *Provider) CreateStore(name string) (err error) {
 
 	db, err := sql.Open("mysql", p.dbURL)
 	if err != nil {
-		return fmt.Errorf("failed to open connection: %w", err)
+		return fmt.Errorf(failureWhileOpeningMySQLConnectionErrMsg, p.dbURL, err)
 	}
 
-	defer func() {
```

Comment:
Is this the "incorrect error handling when closing MySQL connection" bit? Why is it incorrect?

I'm curious because I've used this same idiom before. What was the issue?

Reply:
@llorllale I believe it's just extracted into a function at the end of this file (line 401).

Reply:
@Baha-sk no - the behaviour has changed slightly (no longer wrapping an existing error with `closeErr`) and the PR description mentions fixing "incorrect error handling when closing this connection"... so I wonder what the problem is.

Reply:
Oops... yeah that's a mistake on my part. There's nothing incorrect about it. Will fix...

Reply:
Although I do wonder if there's the possibility of overwriting another (potentially more important) error that happened in the process?

(either way, that wasn't my thought process originally... but it's an interesting point)

Reply:
I reverted that deferred function call and error setting, but I kept the error wrapping. I tried to keep it extracted as an extracted function with the error returned as a pointer argument, I think to make that work, I'd need to make the NewProvider function return an error pointer, which seems weird... let me know if you can think of a better way to do it.

Reply:
> Although I do wonder if there's the possibility of overwriting another (potentially more important) error that happened in the process?
> 
> (either way, that wasn't my thought process originally... but it's an interesting point)

You don't overwrite - you *wrap* one error with the other. Both error messages should show up.